### PR TITLE
[PLT-6440] add code samples for sytem api

### PR DESCRIPTION
**Pull request by cpanato**
the Ping endpoint sample is in another PR.

### CODE REVIEW

File: v4/source/system.yaml [hunk 1]
```diff
@@ -87,6 +117,21 @@
...
+            Client := model.NewAPIv4Client("https://your-mattermost-url.com")
+            Client.Login("email@domain.com", "Password1")
+
+            // GetConfig
+            config, err := Client.GetConfig()
+
+            config.TeamSettings.SiteName = "MyFancyName"
+
+            // UpdateConfig
+            updatedConfig, err := Client.UpdateConfig()
```
Comment (jwilander) [inline v4/source/system.yaml]: This should be `Client.UpdateConfig(config)`, right?

File: v4/source/system.yaml [hunk 2]
```diff
@@ -31,6 +31,16 @@
...
+      x-code-samples:
+        - lang: 'Go'
+          source: |
+            import "github.com/mattermost/platform/model"
+
+            Client := model.NewAPIv4Client("https://your-mattermost-url.com")
+            Client.Login("email@domain.com", "Password1")
+
+            // DatabaseRecycle
+            flag, err := Client.DatabaseRecycle()
```
Comment (jwilander) [inline v4/source/system.yaml]: Can we use `ok` instead of `flag`? I think that's more standard for Go

In all the other samples as well

Comment (jwilander) [inline v4/source/system.yaml]: Also maybe remove the `// DatabaseRecycle` comments, not sure if they're much help since it's just the name of the function

Comment (cpanato) [reply v4/source/system.yaml]: done, sorry for the long delay

File: v4/source/system.yaml (hunk 1 above)
Comment (cpanato) [reply v4/source/system.yaml]: yes true! done.

File: v4/source/system.yaml [hunk 3]
```diff
@@ -48,6 +48,15 @@
...
           $ref: '#/responses/Forbidden'
+      x-code-samples:
+        - lang: 'Go'
+          source: |
+            import "github.com/mattermost/platform/model"
+
+            Client := model.NewAPIv4Client("https://your-mattermost-url.com")
+            Client.Login("email@domain.com", "Password1")
+
+            ok, err := Client.DatabaseRecycle()
```
Comment (jwilander) [inline v4/source/system.yaml]: The `err` should be `resp` here and in the other examples too

Comment (cpanato) [reply v4/source/system.yaml]: done

**Pull request merged**